diff --git a/libvips/foreign/heifsave.c b/libvips/foreign/heifsave.c
--- a/libvips/foreign/heifsave.c
+++ b/libvips/foreign/heifsave.c
@@ -597,27 +597,27 @@ vips_foreign_save_heif_build(VipsObject *object)
 		return -1;
 	}
 
-	error = heif_encoder_set_parameter_integer(heif->encoder,
-		"speed", 9 - heif->effort);
-	if (error.code &&
-		error.subcode != heif_suberror_Unsupported_parameter) {
-		vips__heif_error(&error);
-		return -1;
-	}
-
-	chroma = heif->subsample_mode == VIPS_FOREIGN_SUBSAMPLE_OFF ||
-			(heif->subsample_mode == VIPS_FOREIGN_SUBSAMPLE_AUTO &&
-				heif->Q >= 90)
-		? "444"
-		: "420";
-	error = heif_encoder_set_parameter_string(heif->encoder,
-		"chroma", chroma);
-	if (error.code &&
-		error.subcode != heif_suberror_Unsupported_parameter) {
-		vips__heif_error(&error);
-		return -1;
-	}
+	if (heif->compression != VIPS_FOREIGN_HEIF_COMPRESSION_HEVC) {
+		error = heif_encoder_set_parameter_integer(heif->encoder,
+			"speed", 9 - heif->effort);
+		if (error.code &&
+			error.subcode != heif_suberror_Unsupported_parameter) {
+			vips__heif_error(&error);
+			return -1;
+		}
 
+		chroma = heif->subsample_mode == VIPS_FOREIGN_SUBSAMPLE_OFF ||
+				(heif->subsample_mode == VIPS_FOREIGN_SUBSAMPLE_AUTO &&
+					heif->Q >= 90)
+			? "444"
+			: "420";
+		error = heif_encoder_set_parameter_string(heif->encoder,
+			"chroma", chroma);
+		if (error.code &&
+			error.subcode != heif_suberror_Unsupported_parameter) {
+			vips__heif_error(&error);
+			return -1;
+		}
 #ifdef HAVE_HEIF_ENCODER_PARAMETER_GET_VALID_INTEGER_VALUES
 	for (param = heif_encoder_list_parameters(heif->encoder);
 		*param; param++) {
@@ -677,6 +677,7 @@ vips_foreign_save_heif_build(VipsObject *object)
 			return -1;
 		}
 	}
+	}
 
 	/* TODO .. support extra per-encoder params with
 	 * heif_encoder_list_parameters().
